src/scopeHandler.js

```diff
@@ -84,16 +90,15 @@ export function createScopeHandler(unsafeRec) {
       // todo: allow modifications when target.hasOwnProperty(prop) and it
       // is writable, assuming we've already rejected overlap (see
       // createSafeEvaluatorFactory.factory). This TypeError gets replaced with
-      // target[prop] = value
-      if (objectHasOwnProperty(target, prop)) {
+      // endowments[prop] = value
+      if (prop in endowments) {
```

Comment:
yes... if we copy its descriptors into an internal record, does that mean that inheritance will not work in endowments? that itself is weird, normally APIs will use object.assign() to make a proper copy.

Reply:
@caridy Not sure I follow. Inheritance between what and what?

Reply:
```js
const someOtherEndowment = { x: 1 };
const myEndowments = Object.create(someOtherEndowment, { y: { value: 2 } })
```

if you pass `myEndowments`, should `x` and `y` be used? or only `y`? That's the inheritance question.

Reply:
my argument is that both should be valid endowments, just like any standard API, e.g.: `Object.defineProperty()` where the descriptor object supports such feature.

Reply:
It's interesting, and it all depends on the definition & specifications of "endowments". The drawbacks is tha this forces you to use `in` instead of `hasOwnProperty` in the scope handler `has` trap (as suggested in your other comment about performance).